src/views/User/User.tsx

```diff
@@ -652,46 +652,58 @@ function SelfReportedAccountLinkages({
-        ) {
-            has_other_server = true;
+    const associations = [1, 2, 3] as const;
+    const userAssociations = [];
+    for (const i of associations) {
+        const country = links[`org${i}`];
+        const id = links[`org${i}_id`];
+        const rank = links[`org${i}_rank`];
+        if (country && (id || (rank && rank > 0))) {
+            userAssociations.push({
+                country,
+                id,
+                rank,
+            });
+        }
+    }
+
+    const servers = [
+        ["kgs", _("KGS")],
+        ["igs", _("IGS / PandaNet")],
+        ["dgs", _("DGS")],
+        ["golem", _("Little Golem")],
+        ["wbaduk", _("WBaduk")],
+        ["tygem", _("Tygem")],
+        ["fox", _("Fox")],
+        ["yike", _("Yike Weiqi")],
+        ["goquest", _("GoQuest")],
+        ["badukpop", _("BadukPop")],
+    ] as const;
+    const userServers = [];
+
+    for (const [server, name] of servers) {
+        const id = links[`${server}_username`];
+        const rank = links[`${server}_rank`];
+        if (id || (rank && rank > 0)) {
+            userServers.push({
+                name,
+                id,
+                rank,
+            });
         }
     }
 
     return (
         <div className="SelfReportedAccountLinkages">
-            {has_association && <h3>{_("Associations")}</h3>}
-            <AssociationLink country={links.org1} id={links.org1_id} rank={links.org1_rank} />
-            <AssociationLink country={links.org2} id={links.org2_id} rank={links.org2_rank} />
-            <AssociationLink country={links.org3} id={links.org3_id} rank={links.org3_rank} />
-
-            {has_other_server && <h3>{_("Servers")}</h3>}
-            <ServerLink name={_("KGS")} id={links.kgs_username} rank={links.kgs_rank} />
-            <ServerLink name={_("IGS / PandaNet")} id={links.igs_username} rank={links.igs_rank} />
-            <ServerLink name={_("DGS")} id={links.dgs_username} rank={links.dgs_rank} />
-            <ServerLink
-                name={_("Little Golem")}
-                id={links.golem_username}
-                rank={links.golem_rank}
-            />
-            <ServerLink name={_("WBaduk")} id={links.wbaduk_username} rank={links.wbaduk_rank} />
-            <ServerLink name={_("Tygem")} id={links.tygem_username} rank={links.tygem_rank} />
-            <ServerLink name={_("Fox")} id={links.fox_username} rank={links.fox_rank} />
-            <ServerLink name={_("Yike Weiqi")} id={links.yike_username} rank={links.yike_rank} />
-            <ServerLink name={_("GoQuest")} id={links.goquest_username} rank={links.goquest_rank} />
-            <ServerLink
-                name={_("BadukPop")}
-                id={links.badukpop_username}
-                rank={links.badukpop_rank}
-            />
+            {userAssociations.length > 0 && <h3>{_("Associations")}</h3>}
+            {userAssociations.map((props) => (
+                <AssociationLink {...props} />
```

Comment:
Need to have a `key` prop on this to avoid React complaining at us on the console, either the index or name I reckon

Reply:
I used the spread syntax. The properties are the same as they were (country, id, rank), so react doesn't complain about it.
Should I explicitly define the properties in order to make code more readable maybe?

Reply:
I think

`{userAssociations.map((props) => (
    <AssociationLink key={props.id} {...props} />
))}`

is needed, which is OK since those associations do have unique ids